app/Http/Controllers/ApiTokenController.php

```diff
@@ -4,24 +4,36 @@
 
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
-use App\Models\User;
+use Illuminate\Support\Facades\Gate;
 
 class ApiTokenController extends Controller
 {
     public function showToken(Request $request)
     {
-        return view('showToken', [ 'tokens' => $request->user()->tokens ]);
+        if (!Auth::check()) {
+            return view('welcome');
+        }
+
+        $upload_permission = false;
+
+        if (Gate::allows('upload-import')) {
+            $upload_permission = true;
+        }
+
+        return view('showToken', [ 'tokens' => $request->user()->tokens, 'upload_permission' => $upload_permission ]);
     }
 
     public function createToken(Request $request)
     {
         if (sizeof($request->user()->tokens) > 0) {
-            // token already exists
-            return redirect(route('token'));
+            // token already exists and we want to create a new one
+            // we delete the previous tokens
+            Auth::user()->tokens()->delete();
         }
 
         $token = $request->user()->createToken('apiToken');
-        return view('newToken', [ 'newToken' => $token->plainTextToken ]);
+
+        return redirect('auth/token')->with('flashToken', $token->plainTextToken);
```

Comment:
Nice! Thanks for listening to my crazy talk :D

Reply:
It was easier than I expected. The thing is that in the laravel docs is not clear where you have to put things. But once i figured it out it was all smooth! Thanks!